### PR TITLE
Add Vaisala GLD360-reader.

### PR DESCRIPTION
**Pull request by sjoro**
Add a draft-version reader for Vaisala GLD360 (Global Lightning Detection) data. 

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master -- "*py" | flake8 --diff``
 - [x] Fully documented

### CODE REVIEW

**Comment by sjoro** [comment]
Reader done with test data representing one full day, supported file type is `flashes_20170620.txt`. This is _not_ the official file type from GLD360 network.

**Comment by mraspaud** [comment]
@sjoro What's the status on this ? I the PR feature-complete ? do you plan to add tests ?

**Comment by sjoro** [comment]
@mraspaud from my side it done for now. was thinking about possible test(s) for checking the sensitivity for the input format, but i'd like to get some brainstorming help with that.

**Comment by mraspaud** [comment]
Ok, we can talk about it next week.

**Comment by sjoro** [comment]
Added a test for the file handler.

File: satpy/tests/reader_tests/test_vaisala_gld360.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
```
Comment (sfinkens) [inline satpy/tests/reader_tests/test_vaisala_gld360.py]: This license is not consistent with the one above. IIRC we agreed to remove individual authors from the header.

File: satpy/tests/reader_tests/test_vaisala_gld360.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+"""
+
+import sys
+from io import StringIO
+
+import numpy as np
+
+from satpy.readers.vaisala_gld360 import (
+    VaisalaGLD360TextFileHandler
+)
```
Comment (sfinkens) [inline satpy/tests/reader_tests/test_vaisala_gld360.py]: Parentheses are not necessary

File: satpy/tests/reader_tests/test_vaisala_gld360.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+from satpy.dataset import DatasetID
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
```
Comment (sfinkens) [inline satpy/tests/reader_tests/test_vaisala_gld360.py]: `unittest2` should be added to the package requirements

File: satpy/tests/reader_tests/test_vaisala_gld360.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        filename.close()
+        dataset_id = DatasetID('power')
+        dataset_info = {}
+        result = self.handler.get_dataset(dataset_id, dataset_info).values
+
+        np.testing.assert_allclose(result, expected, rtol=1e-05)
+
+
+def suite():
+    """The test suite for test_scene.
```
Comment (sfinkens) [inline satpy/tests/reader_tests/test_vaisala_gld360.py]: Please replace `test_scene` with `test_vaisala_gld360`

File: satpy/tests/reader_tests/test_vaisala_gld360.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+EXPECTED_POWER = np.array([12.3,  13.2, -31.])
+
+
+class TestVaisalaGLD360TextFileHandler(unittest.TestCase):
+
+    """Test the VaisalaGLD360TextFileHandler."""
+
+    def test_vaisala_gld360(self):
+
+        expected = EXPECTED_POWER
```
Comment (sfinkens) [inline satpy/tests/reader_tests/test_vaisala_gld360.py]: I think `expected = np.array([12.3,  13.2, -31.])` is sufficient here, no need for a constant

File: satpy/readers/vaisala_gld360.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    def start_time(self):
+        return self.data['datetime'].iloc[0]
+
+    @property
+    def end_time(self):
+        return self.data['datetime'].iloc[-1]
+
+    def get_dataset(self, dataset_id, dataset_info):
+        """Load a dataset
+        """
```
Comment (mraspaud) [inline satpy/readers/vaisala_gld360.py]: Docstring's first line should end with a `.`
If it's only one line, the `"""` should be on the same line.

File: satpy/etc/readers/vaisala_gld360.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+  power:
+    name: power
+    sensor: vaisala_gld360
+    resolution: 2000
+    file_type: vaisala_gld360
+    coordinates:
+      - longitude
+      - latitude
+    units: kA
```
Comment (mraspaud) [inline satpy/etc/readers/vaisala_gld360.yaml]: This unit is available from the data file right ? It's good that it's in the yaml file, but I would like to get the actual unit from the data file if possible

File: satpy/etc/readers/vaisala_gld360.yaml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    resolution: 2000
+    file_type: vaisala_gld360
+
+  latitude:
+    name: latitude
+    sensor: vaisala_gld360
+    resolution: 2000
+    file_type: vaisala_gld360
+    standard_name: latitude
+    units: degree
```
Comment (sfinkens) [inline satpy/etc/readers/vaisala_gld360.yaml]: More precisely: `degrees_north`

File: satpy/etc/readers/vaisala_gld360.yaml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+  longitude:
+    name: longitude
+    sensor: vaisala_gld360
+    resolution: 2000
+    file_type: vaisala_gld360
+    standard_name: longitude
+    units: degree
```
Comment (sfinkens) [inline satpy/etc/readers/vaisala_gld360.yaml]: More precisely: `degrees_east`

File: satpy/etc/readers/vaisala_gld360.yaml (hunk 8 above)
Comment (sjoro) [reply satpy/etc/readers/vaisala_gld360.yaml]: CF convention has degree_north, I'll change to that

File: satpy/etc/readers/vaisala_gld360.yaml (hunk 9 above)
Comment (sjoro) [reply satpy/etc/readers/vaisala_gld360.yaml]: Same as above.

File: satpy/etc/readers/vaisala_gld360.yaml (hunk 8 above)
Comment (sfinkens) [reply satpy/etc/readers/vaisala_gld360.yaml]: :+1:

File: satpy/etc/readers/vaisala_gld360.yaml (hunk 7 above)
Comment (sjoro) [reply satpy/etc/readers/vaisala_gld360.yaml]: Added a check that the unit from reader config matches all the units from the file.

**Comment by mraspaud** [comment]
@sjoro can you add this reader to the reader table in the index.rst file (in the doc) ?

**Pull request merged**